### ISSUE TITLE
[BUG] Issue running user guide sample via CLI or project dependency.

### ISSUE THREAD

**Issue by johnson-jesse** [webhintio/hint#3900]
https://webhint.io/docs/user-guide/index.html

When running the CLI command as instructed `npx hint https://example.com` the command fails as follows:

```
% npx hint https://example.com
Using the built-in configuration.
Visit https://webhint.io/docs/user-guide/ to learn how to create your own configuration.
✖ 
Error: No installation found for: ""
    at Object.exports.getInstallationPath (/usr/local/lib/node_modules/hint/node_modules/@hint/connector-puppeteer/dist/src/lib/chromium-finder.js:246:15)
    at PuppeteerConnector.toPuppeteerOptions (/usr/local/lib/node_modules/hint/node_modules/@hint/connector-puppeteer/dist/src/connector.js:55:35)
    at new PuppeteerConnector (/usr/local/lib/node_modules/hint/node_modules/@hint/connector-puppeteer/dist/src/connector.js:37:30)
    at new Engine (/usr/local/lib/node_modules/hint/dist/src/lib/engine.js:33:26)
    at Analyzer.analyze (/usr/local/lib/node_modules/hint/dist/src/lib/analyzer.js:140:27)
    at exports.default (/usr/local/lib/node_modules/hint/dist/src/lib/cli/analyze.js:316:23)
    at async /usr/local/lib/node_modules/hint/dist/src/lib/cli/actions.js:7:20
    at async Object.exports.execute (/usr/local/lib/node_modules/hint/dist/src/lib/cli.js:37:23)
    at async run (/usr/local/lib/node_modules/hint/dist/src/bin/hint.js:44:24)
```

Expected the command to produce some useful feedback on `example.com`. Instead, an error was shown.

Installed version:
```
% node --version
v12.18.0
% npx --version
6.14.4
```

**Comment by johnson-jesse** [comment]
The error message is tracked back to this block of code:
```
const browsers = [Browser.Chrome, Browser.Chromium, Browser.Edge];
    let browserFound = '';
    while (browsers.length > 0 && !browserFound) {
        const br = browsers.shift();
        try {
            browserFound = findBrowserPath(br);
        }
        catch (e) {
        }
    }
    if (!browserFound) {
        throw new Error(ERRORS.NoInstallationFound `${browserFound}`);
    }
```

Is there additional configuration needed for this browser path? I'm relying on webhint defaults since I didn't provide a .hintrc. Do I need to download some browser driver?

**Comment by sarvaje** [comment]
Hi @johnson-jesse,

What OS are you using? Do you have installed any browser?

I'm trying on windows (cmd) and wsl and I can't reproduce it. (I have installed chrome-stable in wsl2 and edge canary in windows).

**Comment by antross** [comment]
@sarvaje Looks like Linux from the file paths.

@johnson-jesse do you happen to have Chrome/Chromium installed? Webhint attempts to use a local Chromium installation by default to avoid bringing down a large browser dependency. There are configuration options for running using JSDOM if Chromium isn't an option, though (it just doesn't currently "fall back" automatically).

**Comment by johnson-jesse** [comment]
@antross @sarvaje Thanks for your replies. I'm on a Mac. Catalina - macOS 10.15.5 (19F101). And I have Google Chrome.app installed in my Applications folder. Version 84.0.4147.89 (Official Build) (64-bit). Do I need to install a specific browser or some driver for Webhint to work with Chrome? What path is Webhint using to look for Chrome? Perhaps that is different on my machine for some reason.

**Comment by sarvaje** [comment]
The code is in this file: https://github.com/webhintio/hint/blob/main/packages/connector-puppeteer/src/lib/chromium-finder.ts

[here](https://github.com/webhintio/hint/blob/main/packages/connector-puppeteer/src/lib/chromium-finder.ts) we are setting some partial paths to look for the browser.

> Do I need to install a specific browser or some driver for Webhint to work with Chrome?

It should work with chrome canary and chrome stable.

**Comment by johnson-jesse** [comment]
@sarvaje @antross Webhint doesn't see my browsers because its grep results are incomplete. None of the combinations are valid paths to any of the installed browsers.

*grep*
```
/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -dump | grep -i '.app$' | awk '{$1=""; print $0}'
```
*produces*
```
com.apple.NanoCalculator.watchkitapp/Contents/MacOS/Google Chrome Canary
id: com.apple.nanocalculator.watchkitapp/Contents/MacOS/Google Chrome Canary
com.apple.dt.UVPreviewAgent-watchOS.watchkitapp/Contents/MacOS/Google Chrome Canary
id: com.apple.dt.uvpreviewagent-watchos.watchkitapp/Contents/MacOS/Google Chrome Canary
ActivityMessagesApp/Contents/MacOS/Google Chrome Canary
com.apple.ActivityMessagesApp/Contents/MacOS/Google Chrome Canary
id: com.apple.activitymessagesapp/Contents/MacOS/Google Chrome Canary
ActivityMessagesApp/Contents/MacOS/Google Chrome Canary
com.apple.Animoji.StickersApp/Contents/MacOS/Google Chrome Canary
id: com.apple.animoji.stickersapp/Contents/MacOS/Google Chrome Canary
com.apple.DocumentsApp/Contents/MacOS/Google Chrome Canary
id: com.apple.documentsapp/Contents/MacOS/Google Chrome Canary
RemoteDesktopApp/Contents/MacOS/Google Chrome Canary
com.apple.siri.parsec.HashtagImagesApp/Contents/MacOS/Google Chrome Canary
id: com.apple.siri.parsec.hashtagimagesapp/Contents/MacOS/Google Chrome Canary
com.apple.ScriptMenuApp/Contents/MacOS/Google Chrome Canary
id: com.apple.scriptmenuapp/Contents/MacOS/Google Chrome Canary
pv-ffd3889c6aab40, NOTIFICATION#:com.apple.ScriptMenuApp/Contents/MacOS/Google Chrome Canary
com.apple.stocks.watchapp/Contents/MacOS/Google Chrome Canary
com.apple.webapp/Contents/MacOS/Google Chrome Canary
pv-e323d62e1058e3, NOTIFICATION#:com.apple.webapp/Contents/MacOS/Google Chrome Canary
com.apple.CarPlayApp/Contents/MacOS/Google Chrome Canary
id: com.apple.carplayapp/Contents/MacOS/Google Chrome Canary
pv-ca036e678042aa, NOTIFICATION#:com.apple.CarPlayApp/Contents/MacOS/Google Chrome Canary
com.apple.UIKitSystemApp/Contents/MacOS/Google Chrome Canary
id: com.apple.uikitsystemapp/Contents/MacOS/Google Chrome Canary
com.cisco.webexmeetingsapp/Contents/MacOS/Google Chrome Canary
pv-d5aa6f36663c6e, NOTIFICATION#DE8Y96K9QP:com.cisco.webexmeetingsapp/Contents/MacOS/Google Chrome Canary
Contents/MacOS/RegisterPluginIMApp/Contents/MacOS/Google Chrome Canary
com.apple.iMovieApp/Contents/MacOS/Google Chrome Canary
id: com.apple.imovieapp/Contents/MacOS/Google Chrome Canary
com.mcafee.maStatusMonitorApp/Contents/MacOS/Google Chrome Canary
id: com.mcafee.mastatusmonitorapp/Contents/MacOS/Google Chrome Canary
pv-d41c885e8486ad, NOTIFICATION#GT8P3H7SPW:com.mcafee.maStatusMonitorApp/Contents/MacOS/Google Chrome Canary
com.microsoft.openxml.excel.app/Contents/MacOS/Google Chrome Canary
pv-d56633954d0a29, NOTIFICATION#UBF8T346G9:com.microsoft.openxml.excel.app/Contents/MacOS/Google Chrome Canary
com.microsoft.openxml.excel.app/Contents/MacOS/Google Chrome Canary
pv-d56633954d0a29, NOTIFICATION#UBF8T346G9:com.microsoft.openxml.excel.app/Contents/MacOS/Google Chrome Canary
/Contents/MacOS/Google Chrome Canary
com.apple.NanoCalculator.watchkitapp/Contents/MacOS/Google Chrome
id: com.apple.nanocalculator.watchkitapp/Contents/MacOS/Google Chrome
com.apple.dt.UVPreviewAgent-watchOS.watchkitapp/Contents/MacOS/Google Chrome
id: com.apple.dt.uvpreviewagent-watchos.watchkitapp/Contents/MacOS/Google Chrome
ActivityMessagesApp/Contents/MacOS/Google Chrome
com.apple.ActivityMessagesApp/Contents/MacOS/Google Chrome
id: com.apple.activitymessagesapp/Contents/MacOS/Google Chrome
ActivityMessagesApp/Contents/MacOS/Google Chrome
com.apple.Animoji.StickersApp/Contents/MacOS/Google Chrome
id: com.apple.animoji.stickersapp/Contents/MacOS/Google Chrome
com.apple.DocumentsApp/Contents/MacOS/Google Chrome
id: com.apple.documentsapp/Contents/MacOS/Google Chrome
RemoteDesktopApp/Contents/MacOS/Google Chrome
com.apple.siri.parsec.HashtagImagesApp/Contents/MacOS/Google Chrome
id: com.apple.siri.parsec.hashtagimagesapp/Contents/MacOS/Google Chrome
com.apple.ScriptMenuApp/Contents/MacOS/Google Chrome
id: com.apple.scriptmenuapp/Contents/MacOS/Google Chrome
pv-ffd3889c6aab40, NOTIFICATION#:com.apple.ScriptMenuApp/Contents/MacOS/Google Chrome
com.apple.stocks.watchapp/Contents/MacOS/Google Chrome
com.apple.webapp/Contents/MacOS/Google Chrome
pv-e323d62e1058e3, NOTIFICATION#:com.apple.webapp/Contents/MacOS/Google Chrome
com.apple.CarPlayApp/Contents/MacOS/Google Chrome
id: com.apple.carplayapp/Contents/MacOS/Google Chrome
pv-ca036e678042aa, NOTIFICATION#:com.apple.CarPlayApp/Contents/MacOS/Google Chrome
com.apple.UIKitSystemApp/Contents/MacOS/Google Chrome
id: com.apple.uikitsystemapp/Contents/MacOS/Google Chrome
com.cisco.webexmeetingsapp/Contents/MacOS/Google Chrome
pv-d5aa6f36663c6e, NOTIFICATION#DE8Y96K9QP:com.cisco.webexmeetingsapp/Contents/MacOS/Google Chrome
Contents/MacOS/RegisterPluginIMApp/Contents/MacOS/Google Chrome
com.apple.iMovieApp/Contents/MacOS/Google Chrome
id: com.apple.imovieapp/Contents/MacOS/Google Chrome
com.mcafee.maStatusMonitorApp/Contents/MacOS/Google Chrome
id: com.mcafee.mastatusmonitorapp/Contents/MacOS/Google Chrome
pv-d41c885e8486ad, NOTIFICATION#GT8P3H7SPW:com.mcafee.maStatusMonitorApp/Contents/MacOS/Google Chrome
com.microsoft.openxml.excel.app/Contents/MacOS/Google Chrome
pv-d56633954d0a29, NOTIFICATION#UBF8T346G9:com.microsoft.openxml.excel.app/Contents/MacOS/Google Chrome
com.microsoft.openxml.excel.app/Contents/MacOS/Google Chrome
pv-d56633954d0a29, NOTIFICATION#UBF8T346G9:com.microsoft.openxml.excel.app/Contents/MacOS/Google Chrome
/Contents/MacOS/Google Chrome
```

**Comment by johnson-jesse** [comment]
@sarvaje @antross I have a fix for y'all or whoever has access to create branches.

* In the grep portion I removed case insensitive flag. Do we really care about finding App? or APP? etc? This is actually non breaking and can be left in.
* Also in grep, I removed the $. As seen below, the path doesn't end in .app. So this entry will never get included if we use $.
* In the awk portion, I updated the print to provide us only index 2 and 3 which turns into `/Applications/Google Chrome.app`

`path:               /Applications/Google Chrome.app (0x2134)`

```
const lines = execSync(
        `${LSREGISTER} -dump` +
        ` | grep '\.app'` + // eslint-disable-line
        ` | awk '{print $2,$3}'`)
        .toString()
        .split(newLineRegex);
```
